### DISCUSSION TITLE
How to either avoid using NEXTAUTH_URL or make it dynamic?

### DISCUSSION THREAD

**Discussion by bnssoftware** [nextauthjs/next-auth#4798] [Help]
### Question 💬

We run a SaaS and offer custom domains.  Our base deployment is account.our-domain.com and we have a customer such as account.customer-domain.com which points to account.our-domain.com.

Since NextAuth seems to require a hard coded value for NEXTAUTH_URL, we can't get this to work.

Is there any way to tell NextAuth to just use the current host name?

### How to reproduce ☕️

Na

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by vincent-thomas** [comment]
> ### Question 💬
> 
> We run a SaaS and offer custom domains.  Our base deployment is account.our-domain.com and we have a customer such as account.customer-domain.com which points to account.our-domain.com.
> 
> Since NextAuth seems to require a hard coded value for NEXTAUTH_URL, we can't get this to work.
> 
> Is there any way to tell NextAuth to just use the current host name?
> 
> ### How to reproduce ☕️
> 
> Na
> 
> ### Contributing 🙌🏽
> 
> No, I am afraid I cannot help regarding this

You can set this in the next-auth url config. Maybe you dont need to hard-code it, bc you can maybe fetch some url in js?
